services/app-web/src/common-code/ContractType.ts

```diff
@@ -0,0 +1,68 @@
+import { Contract } from '../gen/gqlClient'
+
+const isContractOnly = (contract: Contract): boolean => {
```

Comment:
@macrael your comment here feels like it's raising a higher level data validation and types issue (rather than something about small helper functions). It seems to me a little out of scope for Linked Rate. Can we land this work and discuss in RSP?

>I might even suggest that you have that function throw an error if there's neither a draftRevision nor a packageSubmission. 

Want to flag that throwing errors from these types of helper functions means we make the lower level React components handling these helpers have to have error states. This introduces error handling complexity to small child components (right now this is just used in `FileRow`, what would that small chunk of UI display when theres an error for this?). This type of error show be handled way before any page mounts its children, imo at the API layer or right as we exit the API. It would be a higher level error if we had data lacking both a draft Revision AND a package submission... this helper shouldn't even be called.

Reply:
You both have good points. I just pushed up changes to abstract out the rev check to its own function without throwing any errors. Let me know if more changes are needed

Reply:
pearl, I like your fix where we don't check if `contract.packageSubmissions[0]` exists we just return it. This implicitly does what I suggested! Now we'll get a '.submissionType does not exist on undefined' error if we ever fuck this up and return a malformed Contract. Still not quite as clean as it could be but I like that we aren't using null coalescing the same way in the body of the components. I think you could implement something similar for rates to not have to have that ! assertion. 

My issue with this was that instead we had silent error handling. All of our code for displaying bits of form data are using ? null coalescing to catch the error case of if we have a contract with no draft revision nor package submission and silently displaying nothing in that case. That means if we do manage to mess up a contract and send it like that to this component we will have no idea that we did, and we potentially introduce other errors down the line because we're allowing formData to be null when in truth this component really expects formData to exist.